tests/system/data_sources/test_bigquery.py

```diff
@@ -340,15 +351,36 @@ def test_cli_find_tables():
     assert isinstance(tables_json, str)
     assert STRING_MATCH_RESULT in tables_json
 
-    # _remove_bq_conn()
-
 
 def _store_bq_conn():
     parser = cli_tools.configure_arg_parser()
     mock_args = parser.parse_args(BQ_CONN_ARGS)
     main.run_connections(mock_args)
 
 
-# def _remove_bq_conn():
-#     file_path = cli_tools._get_connection_file(BQ_CONN_NAME)
-#     os.remove(file_path)
+def test_random_row_query_builder():
+    bq_client = clients.get_data_client(BQ_CONN)
+    row_query_builder = random_row_builder.RandomRowBuilder(["station_id"], 10)
+    query = row_query_builder.compile(
+        bq_client, "bigquery-public-data.new_york_citibike", "citibike_stations"
+    )
+
+    random_rows = bq_client.execute(query)
+    print(random_rows)
+    print(random_rows["station_id"].values)
+    print(query.compile())
+
+    assert query.compile() == EXPECTED_RANDOM_ROW_QUERY
+    assert len(random_rows["station_id"]) == 10
+    assert list(random_rows["station_id"]) != [
```

Comment:
How do we know the random rows will not select the following station_ids?

Reply:
there are 5000 rows and we randomly select 10 in a random order

so odds this fails randomly are roughly 1/10E36